code/player/Inventory.cs

```diff
@@ -76,72 +73,20 @@ public bool IsCarryingType(Type t)
             return List.Any(x => x.GetType() == t);
         }
 
-        public int NormalizeSlotIndex(int index, int maxIndex)
-        {
-            return index > maxIndex ? 0 : (index < 0 ? maxIndex : index);
-        }
-
-        /// <summary>
-        /// Returns the next inventory slot. If a `HoldType` is given and the corresponding key was pressed, the index of a different carriable entity with
-        /// the same slot will be returned.
-        /// </summary>
-        /// <param name="holdType">Optional. If not set, the next carriable entity in the list will be returned (no matter the slot)</param>
-        /// <returns>
-        /// -1: No selection / carriable entity switch should proceed,
-        /// >0: The index of the carriable entity in the inventory list that should get selected
-        /// </returns>
-        public int GetNextSlot(HoldType holdType = 0)
+        public override bool Drop(Entity entity)
         {
-            int inventorySize = Count();
-
-            if (inventorySize < 1)
+            if (!Host.IsServer || !Contains(entity) || entity is ICarriableItem item && !item.CanDrop())
             {
-                return -1;
-            }
-
-            int activeSlot = GetActiveSlot();
-
-            if (activeSlot != -1)
-            {
-                int nextIndex = NormalizeSlotIndex(activeSlot + 1, inventorySize - 1); // Get next slot index
-                ICarriableItem nextCarr = List[nextIndex] as ICarriableItem;
-
-                if (holdType != 0)
-                {
-                    if (nextCarr.HoldType != holdType)
-                    {
-                        for (int carrIndex = 0; carrIndex < inventorySize; carrIndex++)
-                        {
-                            ICarriableItem indexCarr = List[carrIndex] as ICarriableItem;
-
-                            if (indexCarr.HoldType == holdType)
-                            {
-                                return carrIndex;
-                            }
-                        }
-                    }
-                    else
-                    {
-                        // if there is no carriable entity with same slot or no slot defined, return the next available carriable entity
-                        return nextIndex;
-                    }
-
-                    return -1;
-                }
+                return false;
             }
 
-            // edge case, if List does not contain the active carriable entity
-            return 0;
-        }
-
-        public override bool Drop(Entity ent)
-        {
-            if (!Host.IsServer || !Contains(ent) || ent is ICarriableItem item && !item.CanDrop())
+            using (Prediction.Off())
             {
-                return false;
+                TTTPlayer player = Owner as TTTPlayer;
+                TTTReborn.Globals.RPCs.ClientOnPlayerCarriableItemDrop(To.Single(player), player, entity);
```

Comment:
This don't have to be a global one if it's never intended to have it synced to other client's as well.